### ISSUE TITLE
bind_rows with tf-objects not working 

### ISSUE THREAD

**Issue by fabian-s** [tidyfun/tidyfun#53]
```r
> f <- data_frame(f = rgp(3))
> f2 <- data_frame(f = rgp(4))
> d <- bind_rows(f, f2)
Warning messages:
1: In bind_rows_(x, .id) :
  Vectorizing 'tfd_reg' elements may not preserve their attributes
2: In bind_rows_(x, .id) :
  Vectorizing 'tfd_reg' elements may not preserve their attributes
> d
# A tibble: 7 x 1
  f         
  <list>    
1 <dbl [51]>
2 <dbl [51]>
3 <dbl [51]>
4 <dbl [51]>
5 <dbl [51]>
6 <dbl [51]>
7 <dbl [51]>

# this works, though:
> d <- rbind(f, f2)
> d
# A tibble: 7 x 2
     id f                                                  
  <int> <S3: tfd_reg>                                      
1     1 [1]: (0.00,-0.113);(0.02,-0.110);(0.04,-0.092); ...
2     2 [2]: (0.00, 0.034);(0.02,-0.132);(0.04,-0.271); ...
3     3 [3]: (0.00, -0.81);(0.02, -0.77);(0.04, -0.75); ...
4     4 [4]: (0.00,  0.36);(0.02,  0.24);(0.04,  0.17); ...
5     5 [5]: (0.00, -0.43);(0.02, -0.41);(0.04, -0.40); ...
6     6 [6]: (0.00, -0.67);(0.02, -0.47);(0.04, -0.26); ...
7     7 [7]: (0.00,   1.5);(0.02,   1.4);(0.04,   1.3); ...

> d <- left_join(f, f) # full_join(f, f) etc....
Joining, by = "f"
 Error in left_join_impl(x, y, by_x, by_y, aux_x, aux_y, na_matches) : 
  Can't join on 'f' x 'f' because of incompatible types (tfd_reg/tfd/tf / tfd_reg/tfd/tf)

# this works, though:
> f <- data_frame(id = 1:3, f = rgp(3))
> f2 <- data_frame(id = 4:7, f = rgp(4))
> full_join(f, f2, by = "id")
# A tibble: 7 x 3
     id f.x                                         f.y                                     
  <int> <S3: tfd_reg>                               <S3: tfd_reg>                           
1     1 [1]: (0.00,-0.113);(0.02,-0.110);(0.04,-0.… [1]: (0.00, NULL);(0.02, NULL);(0.04, N…
2     2 [2]: (0.00, 0.034);(0.02,-0.132);(0.04,-0.… [2]: (0.00, NULL);(0.02, NULL);(0.04, N…
3     3 [3]: (0.00, -0.81);(0.02, -0.77);(0.04, -0… [3]: (0.00, NULL);(0.02, NULL);(0.04, N…
4     4 [4]: (0.00,  NULL);(0.02,  NULL);(0.04,  N… [4]: (0.00, 0.36);(0.02, 0.24);(0.04, 0…
5     5 [5]: (0.00,  NULL);(0.02,  NULL);(0.04,  N… [5]: (0.00,-0.43);(0.02,-0.41);(0.04,-0…
6     6 [6]: (0.00,  NULL);(0.02,  NULL);(0.04,  N… [6]: (0.00,-0.67);(0.02,-0.47);(0.04,-0…
7     7 [7]: (0.00,  NULL);(0.02,  NULL);(0.04,  N… [7]: (0.00,  1.5);(0.02,  1.4);(0.04,  …
```

This might need to wait until https://github.com/r-lib/vctrs exists?
Needs to be documented in the mean time...

**Comment by fabian-s** [comment]
putting this into the vignette -- remove there if fixed here.

**Comment by jeff-goldsmith** [comment]
about joins -- the following works:

```
library(tidyfun)
library(tidyverse)

data(chf_df)

monday_df = filter(chf_df, day == "Monday") %>% 
  select(id, monday_act = activity)
friday_df = filter(chf_df, day == "Friday") %>% 
  select(id, friday_act = activity)

left_join(monday_df, friday_df, by = "id")
```

Is the point of the above that you can't join dataframes based on a functional observation? Obviously that's not ideal, but maybe it's not the most common use-case?

**Comment by fabian-s** [comment]
you're right, of course. 
the much more relvant case works and if you do need to join on `f`, the easy workaround is to create a key based on it first and use that for the join. 
I guess we can close this then?

**Comment by jeff-goldsmith** [comment]
maybe, although `bind_rows` is still broken, right?

**Comment by fabian-s** [comment]
true- i'll edit the title and apply wontfix

**Comment by fabian-s** [comment]
Needs to be fixed in tidyverse/dplyr#2432 , or  see r-lib/vctrs/issues/27

**Comment by fabian-s** [comment]
this has been working for a while now, thanks to @jeff-goldsmith 's heroic `vctrs` effort
